File: RepoTransBench/target_projects/Python/C++/oxsecurity_MaskerLogger/public_tests/test_public_masked_logger.cpp
#include <gtest/gtest.h>
#include <string>
#include <memory>
#include <sstream>
#include <iostream>
#include "../src/masker_formatter.h"
#include "../src/utils.h"

using maskerlogger::MaskerFormatter;
using maskerlogger::MaskerFormatterJson;
using maskerlogger::AbstractMaskedLogger;

struct DummyRecord {
    std::string msg;
    bool apply_mask = true;
    DummyRecord(const std::string& message) : msg(message) {}
};

TEST(TestMaskerLoggerFullPathsPublic, MaskSecretLogicWithNewPattern) {
    AbstractMaskedLogger logger(maskerlogger::get_config_file_path());
    std::smatch match;
    std::regex re("(x)(y+)");
    std::string teststr = "xyyyy";
    std::regex_search(teststr, match, re);

    std::string result = logger._mask_secret("xyyyy abc xyyyy", {match});
    size_t star_count = std::count(result.begin(), result.end(), '*');
    ASSERT_GT(star_count, 0u);
}

TEST(TestMaskerLoggerFullPathsPublic, MaskSensitiveDataNoMatchNewmsg) {
    AbstractMaskedLogger logger(maskerlogger::get_config_file_path());
    DummyRecord rec("totally safe entry");
    logger._mask_sensitive_data(rec);
    ASSERT_EQ(rec.msg, "totally safe entry");
}

TEST(TestMaskerLoggerFullPathsPublic, MaskSensitiveDataWithMatchNewsecret) {
    AbstractMaskedLogger logger(maskerlogger::get_config_file_path());
    DummyRecord rec("\"token\": \"abcd12345efgh\" and secret_key = zyxw");
    logger.redact = 39;
    logger._mask_sensitive_data(rec);
    ASSERT_TRUE(typeid(rec.msg) == typeid(std::string));
    ASSERT_EQ(rec.msg.find("abcd12345efgh"), std::string::npos);
}

TEST(TestMaskerLoggerFullPathsPublic, FormatterFullLogIntegrationPublic) {
    MaskerFormatter formatter("%(levelname)s: %(message)s", maskerlogger::get_config_file_path(), 33);
    std::ostringstream ss;
    std::vector<std::string> messages = {
        "\"another_key\": \"AIzaSoMEoth3rKEY344sdlGh289Ka3dLPd\"",
        "\"AWS_SECRET_THISISFAKE\" and some more.",
        "Datadog token is: 'zyxw9876zyxw9876zyxw9876zyxw9876'",
        "\"pin\": \"5678\""
    };
    for (const auto& msg : messages) {
        DummyRecord rec(msg);
        auto masked = formatter.format(rec);
        ss << masked << std::endl;
    }
    std::string output = ss.str();
    ASSERT_NE(output.find("*"), std::string::npos);
}

TEST(TestMaskerLoggerFullPathsPublic, JsonFormatterLogrecordMaskingPublic) {
    MaskerFormatterJson json_formatter("%(message)s", maskerlogger::get_config_file_path(), 22);
    DummyRecord rec("auth = \"FAKENEWSECRETXYZ\"");
    auto value = json_formatter.format(rec);
    ASSERT_NE(value.find("auth"), std::string::npos);
}

TEST(TestMaskerLoggerFullPathsPublic, MaskerFormatterJsonSkipMaskPublic) {
    MaskerFormatterJson json_formatter("%(message)s", maskerlogger::get_config_file_path(), 22);
    DummyRecord rec("publiclogtext");
    rec.apply_mask = false;
    auto result = json_formatter.format(rec);
    ASSERT_NE(result.find("publiclogtext"), std::string::npos);
}

TEST(TestMaskerLoggerFullPathsPublic, ReprAndStrPublic) {
    MaskerFormatterJson json_formatter("%(message)s", maskerlogger::get_config_file_path(), 22);
    std::string clsName = typeid(json_formatter).name();
    ASSERT_NE(clsName.find("Json"), std::string::npos);
}

File: RepoTransBench/target_projects/Python/Rust/oxsecurity_MaskerLogger/tests/public/test_public_masked_logger.rs
use maskerlogger::masker_formatter::*;
use std::sync::Once;

const TEST_CONFIG_PATH: &str = "maskerlogger/config/gitleaks.toml";
static INIT: Once = Once::new();

#[test]
fn test_mask_secret_logic_with_new_pattern() {
    let logger = AbstractMaskedLogger::new(Some(TEST_CONFIG_PATH), 99);
    let sample = "xyyyy";
    let re = regex::Regex::new(r"(x)(y+)").unwrap();
    let m = re.find(sample).unwrap();
    let masked = logger._mask_secret("xyyyy abc xyyyy", vec![m]);
    assert!(masked.chars().filter(|c| *c == '*').count() > 0);
}

#[test]
fn test_mask_sensitive_data_no_match_newmsg() {
    let mut record = DummyRecord::new("totally safe entry");
    let logger = AbstractMaskedLogger::new(Some(TEST_CONFIG_PATH), 10);
    logger._mask_sensitive_data(&mut record);
    assert_eq!(record.msg, "totally safe entry");
}

#[test]
fn test_mask_sensitive_data_with_match_newsecret() {
    let mut record = DummyRecord::new("\"token\": \"abcd12345efgh\" and secret_key = zyxw");
    let logger = AbstractMaskedLogger::new(Some(TEST_CONFIG_PATH), 39);
    logger._mask_sensitive_data(&mut record);
    assert!(record.msg.is_ascii());
    assert!(!record.msg.contains("abcd12345efgh"));
}

#[test]
fn test_formatter_full_log_integration_public() {
    INIT.call_once(|| {
        env_logger::builder().is_test(true).try_init().ok();
    });
    let formatter = MaskerFormatter::new("%(levelname)s: %(message)s", Some(TEST_CONFIG_PATH), 33);
    let mut record = DummyRecord::new("\"another_key\": \"AIzaSoMEoth3rKEY344sdlGh289Ka3dLPd\"");
    record.msg = formatter.format(&record.msg);
    assert!(record.msg.contains("***") || record.msg.contains("another_key"));
}

#[test]
fn test_json_formatter_logrecord_masking_public() {
    let formatter = MaskerFormatterJson::new("%(message)s", Some(TEST_CONFIG_PATH), 22);
    let rec = DummyRecord::new("auth = \"FAKENEWSECRETXYZ\"");
    let value = formatter.format(&rec.msg, true);
    assert!(value.contains("auth"));
}

#[test]
fn test_maskerformatterjson_skip_mask_public() {
    let formatter = MaskerFormatterJson::new("%(message)s", Some(TEST_CONFIG_PATH), 22);
    let mut rec = DummyRecord::new("publiclogtext");
    rec.apply_mask = false;
    let result = formatter.format(&rec.msg, rec.apply_mask);
    assert!(result.contains("publiclogtext"));
}

#[test]
fn test_repr_and_str_public() {
    let formatter = MaskerFormatterJson::new("%(message)s", Some(TEST_CONFIG_PATH), 22);
    let t = format!("{:?}", std::any::type_name::<MaskerFormatterJson>());
    assert!(t.contains("MaskerFormatterJson"));
}